wikimedia/wikipedia-ios
# Native Editor - Wrap up work

**Phabricator:** 
https://phabricator.wikimedia.org/T348092
https://phabricator.wikimedia.org/T331935

### Notes
This PR adds some final pieces to native editor.

### Test Steps

Note: We can lean on QA for deeper regression testing.

1. Simulate a slow connection, go to editor and confirm you see a loading spinner as wikitext loads.
2. Go to an article. Enter airplane mode, then tap edit button to enter editor.
3. Confirm you see a "No internet connection" error.
4. Force a 500 error in the wikitext fetch via a proxy, confirm you see an error message on load as described in https://phabricator.wikimedia.org/T331935 task description (Server error mock)
5. Go to a protected page, confirm you see a protected page error on load. Confirm text view is not editable.
6. Go to a page with edit notices, confirm edit notice presents as it did on web editor (Earth on EN wiki has an edit notice).
7. Edit an article from a blocked account on Test Wiki, confirm you see a blocked error. Confirm text view is not editable.
8. On find and replace, confirm tapping keyboard "search" and "go" buttons initiate find and replace.
9. Highlight a word on article, confirm you no longer see "Edit" in the contextual menu (tracking fix in https://phabricator.wikimedia.org/T348291)
